From da4e71ace51309c782945dc6f072bec925959af9 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 10 Jul 2017 12:40:49 +0800
Subject: [PATCH 2102/5242] MLK-15932-6 gpu: imx: dpu: common: Introduce
 dpu_vproc_get_h/vscale_cap()

commit  71d758e071e1fbcd08d50afbad4e285614eab243 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch introduces two helpers - dpu_vproc_get_h/vscale_cap().

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |   12 ++++++++++++
 include/video/dpu.h              |    3 +++
 2 files changed, 15 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index 171d00e..5a9d27e 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -532,6 +532,18 @@ bool dpu_vproc_has_vscale_cap(u32 cap_mask)
 }
 EXPORT_SYMBOL_GPL(dpu_vproc_has_vscale_cap);
 
+u32 dpu_vproc_get_hscale_cap(u32 cap_mask)
+{
+	return cap_mask & DPU_VPROC_CAP_HSCALE;
+}
+EXPORT_SYMBOL_GPL(dpu_vproc_get_hscale_cap);
+
+u32 dpu_vproc_get_vscale_cap(u32 cap_mask)
+{
+	return cap_mask & DPU_VPROC_CAP_VSCALE;
+}
+EXPORT_SYMBOL_GPL(dpu_vproc_get_vscale_cap);
+
 #define DPU_UNIT_INIT(dpu, base, unit, name, id, pec_ofs, ofs)		\
 {									\
 	int ret;							\
diff --git a/include/video/dpu.h b/include/video/dpu.h
index 2b4730b..6865a70 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -603,6 +603,9 @@ void layerblend_pixengcfg_clken(struct dpu_layerblend *lb,
 bool dpu_vproc_has_hscale_cap(u32 cap_mask);
 bool dpu_vproc_has_vscale_cap(u32 cap_mask);
 
+u32 dpu_vproc_get_hscale_cap(u32 cap_mask);
+u32 dpu_vproc_get_vscale_cap(u32 cap_mask);
+
 /*
  * to avoid on-the-fly/hot plane resource migration
  * between two display interfaces
-- 
1.7.9.5

